atom/browser/browser_linux.cc

```diff
@@ -33,18 +79,41 @@ void Browser::ClearRecentDocuments() {
 void Browser::SetAppUserModelID(const base::string16& name) {
 }
 
-bool Browser::RemoveAsDefaultProtocolClient(const std::string& protocol,
-                                            mate::Arguments* args) {
-  return false;
-}
-
 bool Browser::SetAsDefaultProtocolClient(const std::string& protocol,
                                          mate::Arguments* args) {
-  return false;
+  return SetDefaultWebClient(protocol);
 }
 
 bool Browser::IsDefaultProtocolClient(const std::string& protocol,
                                       mate::Arguments* args) {
+  base::ThreadRestrictions::AssertIOAllowed();
```

Comment:
This check is dummy, since DCHECKS are disabled currently. But it is not good to do IO operations on the UI thread. I suggest moving these to the `FILE` thread instead. Currently we have not started using the new task scheduler api of chromium, which would make scenarios like this easier to deal with. You can look at how the previous version of `{set/check}DefaultProtocolClient` was implemented in chromium using `FILE` thread https://codereview.chromium.org/2888693003 .